### PR TITLE
 Lastline v2: bad errors in test-module

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31966

## Description
When running test module, the error which appeared was not clear. 

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ShahafBenYakir** [comment]
Add tests

File: Packs/Lastline/ReleaseNotes/1_0_4.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Lastline v2
+- Fixed error message at test-module.
```
Comment (ShahafBenYakir) [inline Packs/Lastline/ReleaseNotes/1_0_4.md:4]:
```suggestion
- Fixed a bug in the **test-module** which failed if the response did had no credentials 

```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 2]
```diff
@@ -69,3 +70,16 @@ def test_get_report_context(path):
...
+def raise_exception():
+    raise DemistoException('error Missing required field \'uuid\'.')
+
+
+def test_credentials_not_part_of_params(mocker):
+    mocker.patch.object(demisto, 'params', return_value={'url': 'testurl.com',
+                                                         'api_key': 'apikey',
+                                                         'api_token': 'apitoken'})
+    mocker.patch.object(demisto, 'command', return_value='test-module')
+    mocker.patch.object(Client, 'get_report', side_effect=raise_exception)
```
Comment (ShahafBenYakir) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:84]: This is not a good way to validate the exception, need to use `with raises` instead. In general, I don't understand what are we testing here. Ping me when you are getting back to this.

**Review by ShahafBenYakir — CHANGES_REQUESTED** [review]
The change you added relates to the changes introduced by @adi88d [here](https://github.com/demisto/content/pull/10121/files?file-filters%5B%5D=). You have still did not resolved the bad message displayed when using bad email and password: 
Settings 2020-12-21 21-04-53

In addition, a documentation addition for the new params is needed as was explained in the issue.

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml [hunk 3]
```diff
@@ -12,17 +12,20 @@ configuration:
   name: api_key
   required: false
   type: 0
-  additionalinfo: Note that whe when credentials are filled with data, the integration use account-based authentication.
+  additionalinfo: Note that whe when credentials are filled with data, the integration
+    use account-based authentication.
```
Comment (glicht) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml:16]:
```suggestion
    uses account-based authentication.
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml [hunk 4]
```diff
@@ -12,17 +12,20 @@ configuration:
   name: api_key
   required: false
   type: 0
-  additionalinfo: Note that whe when credentials are filled with data, the integration use account-based authentication.
+  additionalinfo: Note that whe when credentials are filled with data, the integration
```
Comment (glicht) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml:15]:
```suggestion
  additionalinfo: Note that when credentials are set, the integration
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml [hunk 5]
```diff
@@ -12,17 +12,20 @@ configuration:
...
+    use account-based authentication.
 - display: API Token for accessing Lastline APIs
   name: api_token
   required: false
   type: 4
-  additionalinfo: Note that when credentials are filled with data, the integration use account-based authentication.
+  additionalinfo: Note that when credentials are filled with data, the integration
```
Comment (glicht) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml:21]:
```suggestion
  additionalinfo: Note that when credentials are set, the integration
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml [hunk 6]
```diff
@@ -12,17 +12,20 @@ configuration:
...
+    use account-based authentication.
 - display: Email address
   name: credentials
   required: false
   type: 9
-  additionalinfo: Note that when credentials are filled with data, the integration use account-based authentication.
+  additionalinfo: Note that when credentials are filled with data, the integration
```
Comment (glicht) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml:27]:
```suggestion
  additionalinfo: Note that when credentials are set, the integration
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml [hunk 7]
```diff
@@ -12,17 +12,20 @@ configuration:
...
+    use account-based authentication.
```
Comment (glicht) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.yml:28]:
```suggestion
    uses account-based authentication.
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py (hunk 2 above)
Comment (daryakoval) [reply Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py]: explained to you the test, in our zoom conversation

File: Packs/Lastline/ReleaseNotes/1_0_4.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### Lastline v2
+- Fixed a bug in the **test-module** which failed if no email and username were given.
+
```
Comment (ShahafBenYakir) [inline Packs/Lastline/ReleaseNotes/1_0_4.md:5]:
```suggestion
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.py [hunk 9]
```diff
@@ -4,6 +4,7 @@
 import demistomock as demisto
 from CommonServerPython import *
 from CommonServerUserPython import *
+from datetime import datetime
```
Comment (ShahafBenYakir) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.py:7]: I don't think you need it... you get it from common server python

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 10]
```diff
@@ -69,3 +70,77 @@ def test_get_report_context(path):
...
+def raise_exception():
+    raise DemistoException('Authentication Error.')
+
+
+@pytest.mark.parametrize('params', [{'url': 'testurl.com',
+                                     'credentials': {'identifier': 'identifier',
+                                                     'password': 'password'}},
+                                    {'url': 'testurl.com',
+                                     'username': 'identifier',
+                                     'password': 'password-wrong'},
```
Comment (ShahafBenYakir) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:123]: What is this there no such option in the integration for param `username`. Did you mean to put here token and key?

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 11]
```diff
@@ -69,3 +70,77 @@ def test_get_report_context(path):
...
+    Given:
+        - connect cedentials
+
+    When:
+        - testing connect to integration, using credentials: email and password.
+
+    Then:
+        - validating no error is raised.
+
+    """
```
Comment (ShahafBenYakir) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:136]:
```suggestion
    """
    Given:
        - Case a: wrong user and password
        - Case b: wrong API token and key

    When:
        - Running the integration test module

    Then:
        - Validating the right error message raised

    """
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 12]
```diff
@@ -69,3 +70,77 @@ def test_get_report_context(path):
...
```
Comment (ShahafBenYakir) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:103]:
```suggestion
    Given:
        - Valid credentials

    When:
        - Running the integration test module, using valid credentials: email and password.

    Then:
        - validating no error is raised.

    """
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 13]
```diff
@@ -69,3 +70,77 @@ def test_get_report_context(path):
             json_obj = json_obj.read()
             out_dict = json.loads(json_obj)
         get_report_context(out_dict)
+
+
+def test_credentials_not_part_of_params(mocker):
+    """
```
Comment (ShahafBenYakir) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:76]: Add given

**Review by ShahafBenYakir — COMMENTED** [review]
Pull from master

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.py (hunk 9 above)
Comment (daryakoval) [reply Packs/Lastline/Integrations/Lastline_v2/Lastline_v2.py:7]: removed

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py (hunk 10 above)
Comment (daryakoval) [reply Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:123]: dont understand what you mean, but there is an error at these lines, fixed

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py (hunk 13 above)
Comment (daryakoval) [reply Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:76]: done

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**